botocore/paginate.py

```diff
@@ -472,6 +471,10 @@ def _extract_paging_params(self, kwargs):
             max_items = int(max_items)
         page_size = pagination_config.get('PageSize', None)
         if page_size is not None:
+            if self._pagination_cfg.get('limit_key', None) is None:
+                raise PaginationError(
+                    message="Page size unsupported by the service for this "
```

Comment:
Can we make the error message a little more clear like (mainly the spelling of Page size):
``PageSize parameter is not supported for the pagination interface for this operation``